### DISCUSSION TITLE
Export results - how to extract outgoing queries / build a search object from UI state

### DISCUSSION THREAD

**Discussion by joepio** [algolia/instantsearch#5436] [Q&A] [answered]
Hi there! 

I'm using `react-instantsearch-hooks-web` to build a Map search app, and one of the things I'd like to do is to _export the results_. I think I should do this by taking the current query, setting the results limit as high as possible, iterating over the pages, and then creating a file download client-side. Seems doable, except for the _taking the current query_ part.

**How do I get the JSON object representing the query?**

I tried the [Middleware](https://www.algolia.com/doc/api-reference/widgets/middleware/js/) approach, but that only gives me UI state - not a complete Query body.

I also tried `useQuery`, but that doesn't seem to return the query either (just a object with an empty `items` array?)

I tried `useConnector`, but wasn't quite sure what I should pass to it.

I've browsed the instantsearch codebase for a while, but it's kind of hards to navigate for me. 

Any help is greatly appreciated!

**Comment by Haroenv** [comment, marked as answer]
To get the parameters usable by a search client or browse, you can use the query that would be generated by algoliasearch-helper. To get access to that, get access to the search parameters, transform them to query parameters, and after that do what we want with it (maybe frontend, maybe backend, maybe search, maybe browse).

```js
import React from "react";
import { useInstantSearch } from "react-instantsearch-hooks-web";
import requestBuilder from "algoliasearch-helper/src/requestBuilder";

export function Exporter() {
  const parameters = useInstantSearch();
  return (
    <button
      type="button"
      onClick={() => {
        alert("see logs");
        console.log(
          requestBuilder._getHitsSearchParams(parameters.results._state)
        );
      }}
    >
      export current search parameters
    </button>
  );
}

```

https://codesandbox.io/s/youthful-noyce-duygx3?file=/src/components/Exporter.tsx

Note that this is private API, and may change without breaking change, so you could also copy the actual implementation subset you need in your case.

**Reply by joepio** [reply]
Thanks! :)
